test/vmtester/mock_context.cpp

```diff
@@ -0,0 +1,136 @@
+                        const struct evmc_uint256be* key)
+{
+    (void)result;
+    (void)context;
+    (void)address;
+    (void)key;
+}
+
+static enum evmc_storage_status set_storage(struct evmc_context* context,
+                                            const struct evmc_address* address,
+                                            const struct evmc_uint256be* key,
+                                            const struct evmc_uint256be* value)
+{
+    (void)context;
+    (void)address;
+    (void)key;
+    (void)value;
+    return EVMC_STORAGE_UNCHANGED;
+}
+
+static void get_balance(struct evmc_uint256be* result,
+                        struct evmc_context* context,
+                        const struct evmc_address* address)
+{
+    *result = balance(context, address);
+}
+
+static size_t get_code_size(struct evmc_context* context, const struct evmc_address* address)
+{
+    (void)context;
+    (void)address;
+    return 0;
+}
+
+static void get_code_hash(struct evmc_uint256be* result,
+                          struct evmc_context* context,
+                          const struct evmc_address* address)
+{
+    (void)result;
+    (void)context;
+    (void)address;
+}
+
+static size_t copy_code(struct evmc_context* context,
+                        const struct evmc_address* address,
+                        size_t code_offset,
+                        uint8_t* buffer_data,
+                        size_t buffer_size)
+{
+    (void)context;
+    (void)address;
+    (void)code_offset;
+    (void)buffer_data;
+    (void)buffer_size;
+    return 0;
+}
+
+static void selfdestruct(struct evmc_context* context,
+                         const struct evmc_address* address,
+                         const struct evmc_address* beneficiary)
+{
+    (void)context;
+    (void)address;
+    (void)beneficiary;
+}
+
+static void call(struct evmc_result* result,
+                 struct evmc_context* context,
+                 const struct evmc_message* msg)
+{
+    (void)context;
+    (void)msg;
+    result->status_code = EVMC_FAILURE;
+}
+
+static void get_tx_context(struct evmc_tx_context* result, struct evmc_context* context)
+{
+    (void)result;
+    (void)context;
+}
+
+static void get_block_hash(struct evmc_uint256be* result,
+                           struct evmc_context* context,
+                           int64_t number)
+{
+    (void)result;
+    (void)context;
+    (void)number;
+}
+
+/// EVM log callback.
+static void emit_log(struct evmc_context* context,
+                     const struct evmc_address* address,
+                     const uint8_t* data,
+                     size_t data_size,
+                     const struct evmc_uint256be topics[],
+                     size_t topics_count)
+{
+    (void)context;
+    (void)address;
+    (void)data;
+    (void)data_size;
+    (void)topics;
+    (void)topics_count;
+}
+
+extern const struct evmc_context_fn_table mock_context_fn_table = {
```

Comment:
In the future we could extend this to have a factory which instantiates a context with certain values.